src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -99,7 +99,7 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
   auto lease_client = GetOrConnectLeaseClient(raylet_address);
   TaskSpecification &resource_spec = it->second.front();
   TaskID task_id = resource_spec.TaskId();
-  RAY_CHECK_OK(lease_client->RequestWorkerLease(
+  auto status = lease_client->RequestWorkerLease(
```

Comment:
Turns out this can fail if the remote actor is dead.

Reply:
Yes, that's true.

Reply:
Do we know exactly what the conditions are for it to fail? My understanding is that it *sometimes* fails when the remote connection is down?

Reply:
Yeah, my guess is if the connection is already known to be unhealthy it will fail immediately.

Reply:
yes I think that's the case.